engine/link/link.asm

```diff
@@ -1882,23 +1882,29 @@ LinkTrade:
 	call LoadTradeScreenBorder
 	call SetTradeRoomBGPals
 	farcall Link_WaitBGMap
-	ld b, $1
+
+; Check if either of the pokemon sent was a Mew or Celebi, and send a different
+; byte depending on that. Presumably this would've been some prevention against
+; illicit trade machines, but it doesn't seem like a very effective one.
+; Removing this code breaks link compatibility with the vanilla games, but has
+; otherwise no consequence.
+	ld b, 1
 	pop af
 	ld c, a
 	cp MEW
 	jr z, .loop
```

Comment:
The `.loop` makes this slightly confusing since no looping is going on. How about a `.set_b` label at the end of this block, coinciding with the `.loop` below?

Reply:
Looping is going on, though. If the value being sent by the other device doesn't match the `b` decided here, it keeps looping.

Reply:
What I meant is the `jr z, .loop` lines in this block are not going backwards to re-iterate a loop; they're going forwards to the beginning of a loop below. Or rather, to the end of the block that sets `b`, and right after it is the beginning of the block that loops and compares the received value with `b`.

Reply:
May as well also comment above the `cp LINK_TIMECAPSULE` noting that Gen 1 games don't send such a byte for Mew.

Reply:
> What I meant is the jr z, .loop lines in this block are not going backwards to re-iterate a loop; they're going forwards to the beginning of a loop below.

This is a rather common pattern you'll see across the codebase, except with names other than `.loop`. Would `.send_checkbyte` work?

> May as well also comment above the cp LINK_TIMECAPSULE noting that Gen 1 games don't send such a byte for Mew.

"Removing this code breaks link compatibility with **other gen2 games**"?

Reply:
Sure, it's a long function and some of those `.asm_XXXX` labels are probably loops too, so `.send_checkbyte` is more explicit.